src/perf_counters_c.c

```diff
@@ -0,0 +1,78 @@
+/* -*- C -*-
+ *
+ * Copyright 2011 Sandia Corporation. Under the terms of Contract
+ * DE-AC04-94AL85000 with Sandia Corporation, the U.S.  Government
+ * retains certain rights in this software.
+ *
+ * Copyright (c) 2018 Intel Corporation. All rights reserved.
+ * This software is available to you under the BSD license.
+ *
+ * This file is part of the Sandia OpenSHMEM software package. For license
+ * information, see the LICENSE file in the top level directory of the
+ * distribution.
+ *
+ */
+
+#include "shmem_internal.h"
+#include "transport.h"
+
+#ifdef ENABLE_PROFILING
+#include "pshmem.h"
+
+#pragma weak shmemx_pcntr_get_pending_put = pshmemx_pcntr_get_pending_put
+#define shmemx_pcntr_get_pending_put pshmemx_pcntr_get_pending_put
+
+#pragma weak shmemx_pcntr_get_pending_get = pshmemx_pcntr_get_pending_get
+#define shmemx_pcntr_get_pending_get pshmemx_pcntr_get_pending_get
+
+#pragma weak shmemx_pcntr_get_completed_put = pshmemx_pcntr_get_completed_put
+#define shmemx_pcntr_get_completed_put pshmemx_pcntr_get_completed_put
+
+#pragma weak shmemx_pcntr_get_completed_get = pshmemx_pcntr_get_completed_get
+#define shmemx_pcntr_get_completed_get pshmemx_pcntr_get_completed_get
+
+#pragma weak shmemx_pcntr_get_completed_target = pshmemx_pcntr_get_completed_target
+#define shmemx_pcntr_get_completed_target pshmemx_pcntr_get_completed_target
+
+#endif /* ENABLE_PROFILING */
+
+void SHMEM_FUNCTION_ATTRIBUTES 
+shmemx_pcntr_get_pending_put(shmem_ctx_t ctx, uint64_t *cntr_value)
+{
+    SHMEM_ERR_CHECK_INITIALIZED();
+    *cntr_value = shmem_transport_get_pending_put_cntr((shmem_transport_ctx_t *) ctx);
+    return;
+}
+
+void SHMEM_FUNCTION_ATTRIBUTES 
+shmemx_pcntr_get_pending_get(shmem_ctx_t ctx, uint64_t *cntr_value)
+{
+    SHMEM_ERR_CHECK_INITIALIZED();
+    *cntr_value = shmem_transport_get_pending_get_cntr((shmem_transport_ctx_t *) ctx);
```

Comment:
Could you rename the transport API routines to match the corresponding shmemx routines?  E.g., `shmemx_pcntr_get_pending_get` should map to `shmem_transport_pcntr_get_pending_get`.

Reply:
Thanks @jdinan. Will do. 
Do you think the APIs for the two new counters you introduced in #704 will be handy?